File: backend/find_factory.py
import os
from web3 import Web3
from dotenv import load_dotenv

load_dotenv()

w = Web3(Web3.HTTPProvider(os.getenv('ALCHEMY_RPC_URL')))
deployer = "0xa30A689ec0F9D717C5bA1098455B031b868B720f"

print(f"Connected: {w.is_connected()}")
nonce = w.eth.get_transaction_count(deployer)
print(f"Current nonce: {nonce}")

# Calculate contract addresses for recent nonces
import rlp

def get_contract_address(sender, nonce):
    sender_bytes = bytes.fromhex(sender[2:])
    raw = rlp.encode([sender_bytes, nonce])
    addr = w.keccak(raw)[12:]
    return w.to_checksum_address(addr.hex())

print("\nRecent deployments:")
for n in range(60, nonce + 1):
    addr = get_contract_address(deployer, n)
    code = w.eth.get_code(addr)
    has_code = len(code) > 2
    print(f"  Nonce {n}: {addr} {'[CONTRACT]' if has_code else ''}")

File: scripts/findFactory.js
const { ethers } = require("hardhat");

async function main() {
    const deployer = "0xa30A689ec0F9D717C5bA1098455B031b868B720f";

    console.log("Checking deployer:", deployer);
    const nonce = await ethers.provider.getTransactionCount(deployer);
    console.log("Current nonce:", nonce);

    console.log("\nChecking recent contract addresses:");
    for (let n = 60; n < nonce; n++) {
        // ethers v6: getCreateAddress
        const addr = ethers.getCreateAddress({ from: deployer, nonce: n });
        const code = await ethers.provider.getCode(addr);
        const hasCode = code.length > 2;
        console.log(`  Nonce ${n}: ${addr} ${hasCode ? '[CONTRACT]' : '[EOA]'}`);
    }

    // Check nonce 61 and 62 specifically (likely factory)
    console.log("\n=== Most Likely Factory Addresses ===");
    for (let n = 61; n <= 62; n++) {
        const addr = ethers.getCreateAddress({ from: deployer, nonce: n });
        console.log(`Nonce ${n}: ${addr}`);

        try {
            const Factory = await ethers.getContractAt("TechneAccountFactory", addr);
            const impl = await Factory.implementation();
            console.log(`  -> This is the FACTORY! Implementation: ${impl}`);
        } catch (e) {
            console.log(`  -> Not the factory (${e.message.slice(0, 50)}...)`);
        }
    }
}

main().catch(console.error);
